source/file.c

```diff
@@ -23,6 +28,11 @@ FILE *aws_fopen(const char *file_path, const char *mode) {
 }
 
 int aws_byte_buf_init_from_file(struct aws_byte_buf *out_buf, struct aws_allocator *alloc, const char *filename) {
+    if (!filename || strlen(filename) == 0) {
```

Comment:
This seems redundant given the addition to aws_fopen

Reply:
I modified this so now it checks to see if `errno` is `0` after calling `aws_fopen`, which should mean that the file returned is `null` and, if it is, then it will return `AWS_ERROR_FILE_INVALID_PATH` rather than `aws_translate_and_raise_io_error(errno)`. 👍